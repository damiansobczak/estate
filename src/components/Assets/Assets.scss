@import "../../assets/styles/utils/variables";
@import "../../assets/styles/utils/mixins";

.assets {
  background-color: $color-neutral-100;
  padding: 50px 0;
  margin-top: 40px;

  &--empty {
    text-align: center;
  }

  &__title {
    margin-bottom: 10px;
  }

  &__item {
    display: flex;
    flex-direction: column;
    text-align: left;
    position: relative;
    padding-right: 20px;
    margin: 5px 0;

    &:after {
      content: "";
      position: absolute;
      right: 20px;
      top: 0;
      bottom: 0;
      height: 100%;
      width: 1px;
      background-color: rgba($color-neutral-300, 0.4);
    }

    &:last-child {
      &:after {
        background-color: transparent;
      }
    }

    @include breakpoint-down(md) {
      &:nth-child(even) {
        &:after {
          background-color: transparent;
        }
      }
    }
  }

  &__name {
    color: $color-text-secondary;
    font-weight: 600;
    margin-bottom: 8px;
  }

  &__desc {
    color: $color-text-primary;
    font-weight: 700;
  }
}
