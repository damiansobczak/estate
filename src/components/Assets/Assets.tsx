import React from "react";
import "./Assets.scss";
import { AssetsProps } from "./Interfaces/AssetsProps";

class Assets extends React.Component<AssetsProps> {
  render() {
    if (this.props.assets && this.props.assets.length > 0) {
      return (
        <section className="assets">
          <div className="container">
            <div className="columns is-multiline is-mobile">
              {this.props.assets.map(asset => (
                <div className="assets__item column is-6-mobile is-4-tablet is-2-desktop" key={asset.id}>
                  <div className="assets__name">{asset.name.trim() ? asset.name : "-"}</div>
                  <div className="assets__desc">{asset.desc.trim() ? asset.desc : "-"}</div>
                </div>
              ))}
            </div>
          </div>
        </section>
      );
    } else {
      return (
        <section className="assets assets--empty">
          <div className="container">
            <div className="columns is-multiline is-mobile">
              <div className="column">
                <h4 className="assets__title">Przepraszamy, brak danych.</h4>
                <div className="columns">
                  <div className="column is-6-tablet is-offset-3-tablet">
                    <p className="assets__subtitle">Obecnie nie posiadamy szczegółów odnośnie tej oferty. Skontaktuj się z naszym biurem lub agentem telefonicznie.</p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </section>
      );
    }
  }
}

export default Assets;
