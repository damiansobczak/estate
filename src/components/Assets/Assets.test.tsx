import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Assets from "./Assets";

const assets = [
  { id: 1, name: "Łazienka", desc: "Prysznic, pralka" },
  { id: 2, name: "Pokoje", desc: "2, TV, stół, sofa" },
  { id: 3, name: "Kuchnia", desc: "Kuchenka, czajnik" },
  { id: 4, name: "Taras", desc: "6m2, zastawa" },
  { id: 5, name: "Parking", desc: "Miejsce, szlaban" },
  { id: 6, name: "Metraż", desc: "26m2, taras" }
];

const assetsEmpty = [{ id: 1, name: " ", desc: "Prysznic, pralka" }];

const assetsEmpty2 = [{ id: 1, name: "", desc: "Prysznic, pralka" }];

describe("<Assets />", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Assets assets={assets} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders props correctly", () => {
    const wrapper = shallow(<Assets assets={assets} />);
    expect(wrapper.find(".assets__item").length).toEqual(6);
  });

  it("renders props correctly", () => {
    const wrapper = shallow(<Assets assets={assets} />);
    const arr = wrapper.find(".assets__item");
    const arr2 = arr.forEach(el => el.text());
    expect(arr2.contains("Łazienka")).toBeTruthy();
  });

  it("renders - when object property not provided with space", () => {
    const wrapper = shallow(<Assets assets={assetsEmpty} />);
    expect(
      wrapper
        .find(".assets__item")
        .text()
        .split("")
        .includes("-")
    ).toBeTruthy();
  });

  it("renders - when object property not provided", () => {
    const wrapper = shallow(<Assets assets={assetsEmpty2} />);
    expect(
      wrapper
        .find(".assets__item")
        .text()
        .split("")
        .includes("-")
    ).toBeTruthy();
  });

  it("renders error when assets empty", () => {
    const wrapper = shallow(<Assets assets={[]} />);
    expect(wrapper.find(".assets__title").text()).toEqual("Przepraszamy, brak danych.");
  });
});
