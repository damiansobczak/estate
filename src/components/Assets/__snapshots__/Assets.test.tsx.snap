// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Assets /> renders correctly 1`] = `
<section
  className="assets"
>
  <div
    className="container"
  >
    <div
      className="columns is-multiline is-mobile"
    >
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Łazienka
        </div>
        <div
          className="assets__desc"
        >
          Prysznic, pralka
        </div>
      </div>
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Pokoje
        </div>
        <div
          className="assets__desc"
        >
          2, TV, stół, sofa
        </div>
      </div>
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Kuchnia
        </div>
        <div
          className="assets__desc"
        >
          Kuchenka, czajnik
        </div>
      </div>
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Taras
        </div>
        <div
          className="assets__desc"
        >
          6m2, zastawa
        </div>
      </div>
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Parking
        </div>
        <div
          className="assets__desc"
        >
          Miejsce, szlaban
        </div>
      </div>
      <div
        className="assets__item column is-6-mobile is-4-tablet is-2-desktop"
      >
        <div
          className="assets__name"
        >
          Metraż
        </div>
        <div
          className="assets__desc"
        >
          26m2, taras
        </div>
      </div>
    </div>
  </div>
</section>
`;
